File: practice/algorithms/implementation/grading_students/GradingStudents.scala
object GradingStudents extends App {
  val grades = Source.stdin.getLines().toList.tail.map(_.toInt)
  val finalGrades = grades.map(professorGradeRule)

  println(finalGrades.mkString("\n"))

  def professorGradeRule(grade: Int): Int = {
    val multiply5 = grade / 5
    val roundBy5 = if(grade % 5 == 0) multiply5 * 5 else (multiply5 + 1) * 5
    if(roundBy5 - grade < 3 && grade >= 38) roundBy5 else grade
  }
}

File: practice/algorithms/implementation/grading_students/GradingStudents.java
import java.util.*;

public class GradingStudents {

    private static int professorGradeRule(int grade) {
        int multiply5 = grade / 5;
        int roundBy5 = (grade % 5 == 0) ? (multiply5 * 5) : ((multiply5 + 1) * 5);
        if(roundBy5 - grade < 3 && grade >= 38) {
            return roundBy5;
        }
        return grade;
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int tests = Integer.parseInt(stdin.nextLine());
        for(int i = 0; i < tests; i++) {
            int grade = Integer.parseInt(stdin.nextLine());
            int finalGrade = professorGradeRule(grade);
            System.out.println(finalGrade);
        }
        stdin.close();
    }
}